pkg/localgenerator/localfs.go

```diff
@@ -0,0 +1,16 @@
+package localgenerator
+
+import (
+	"io/ioutil"
+	"os"
+)
+
+type LocalGenerator struct{}
+
+func (l LocalGenerator) ReadDir(dir string) ([]os.FileInfo, error) {
+	return ioutil.ReadDir(dir)
+}
+
+func (l LocalGenerator) ReadFile(filename string) ([]byte, error) {
+	return ioutil.ReadFile(filename)
```

Comment:
This should not work because we use `github.IsNotFound`. You'd need to rearrange things a bit. I can help you with that.

Reply:
Isn't this a wrong interface then? I mean, it compiles and works for my case. But if it is not expected to work - then it should fail on the compilation, no?

Reply:
Compilation is irrelevant. I'm not sure what would happen if one of the optional files is missing. Then `github.IsNotFound` won't catch that.

See: https://github.com/giantswarm/config-controller/blob/2cc199fe52cc334dc49361b5851abda885ee12bd/pkg/generator/error.go#L24

And grep https://github.com/giantswarm/config-controller/blob/2cc199fe52cc334dc49361b5851abda885ee12bd/pkg/generator/generator.go for `IsNotFound`.

Reply:
I understand this. I meant that having implementation-specific error in interface package sounds wrong to me

Reply:
I also think it's wrong. That's why I opened this https://github.com/giantswarm/giantswarm/issues/14358